server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -377,6 +383,8 @@ public void run() {
     } else {
       authKeyWatcher = null;
     }
+    MAX_WRITE_THREADS_DEFAULT =
+        getServerConfig().getConfiguration().getDefaultCount(Property.TSERV_WRITE_THREADS_MAX);
```

Comment:
I don't understand setting this. What's the difference between this new method you added and `DefaultConfiguration.getInstance().getCount(Property.TSERV_WRITE_THREADS_MAX)`?

Reply:
This gets the default value. The other method gets the configured value. I didn't want to hardcode 0 because we could change the default value. Granted, here it isn't that big of deal because we probably won't change the default value.

Reply:
I'm pretty sure they are both getting the default value for the same property... but one is doing so using a new method that was added unnecessarily. And, assigning it here isn't even necessary at all, because we need to compare to `0`... `0` semantically means "unlimited" not `default`. If the default was `100`, that doesn't mean we should have unlimited threads... that would be *very* unusual semantics.